app/scripts/directives/oscPersistentVolumeClaim.js

```diff
@@ -26,7 +27,24 @@ angular.module("openshiftConsole")
         }];
         scope.claim.selectedLabels = [];
         DataService.list({group: 'storage.k8s.io', resource: 'storageclasses'}, {}, function(storageClasses) {
-          scope.storageClasses = storageClasses.by('metadata.name');
+          scope.storageClasses = _.map(storageClasses.by('metadata.name'));
+          var annotation = $filter('annotation');
+          scope.defaultStorageClass = _.find(scope.storageClasses, function(storageClass) {
+            return annotation(storageClass, 'storageclass.beta.kubernetes.io/is-default-class') === 'true';
+          });
+          if (!scope.defaultStorageClass)  { //if there is no default, set a no storage class option
+            var noclass = {
+              metadata: {
+                name: "\u0020No Storage Class",  //unicode char to make this first alphabetically
```

Comment:
This will perform better as well since you're not resorting the list on every Angular digest loop, just once when we request it.